### PR TITLE
feat: add frontend option

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: examples/existing_private_network/vars.auto.tfvars [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+subnet_name             = "weka-subnet-0"
+subnet_delegation       = "10.0.2.0/25"
+cluster_name            = "poc"
+private_network         = true
+install_weka_url        = "https://wekadeploytars.blob.core.windows.net/tars/weka-4.1.0.71.tar"
+instance_type           = "Standard_L8s_v3"
+set_obs_integration     = true
+tiering_ssd_percent     = 20
+cluster_size            = 6
+add_frontend_containers = true
```
Comment (assafgi) [inline examples/existing_private_network/vars.auto.tfvars:13]: not needed, we will have amount of zero

File: function-app/code/go.sum [hunk 2]
```diff
@@ -57,6 +57,50 @@ github.com/rs/zerolog v1.29.0/go.mod h1:NILgTygv/Uej1ra5XxGf82ZFSLk58MFGAUS2o6us
...
+github.com/weka/go-cloud-lib v0.0.0-20230622120735-7d74e9181a96 h1:ZwS08eiHt1KFa+9Gu7LFQxL/LSwLsiB5xRTbAMxe2y8=
+github.com/weka/go-cloud-lib v0.0.0-20230622120735-7d74e9181a96/go.mod h1:lT0kVeV0d4HznnITUHbHN6JunpqyL4q9McAYr78YNm4=
+github.com/weka/go-cloud-lib v0.0.0-20230622122219-feb1c51b50d6 h1:4gDlRBV5VWDYiRaHd72fj/mKlooZ2IuuIZZvGegZZH4=
+github.com/weka/go-cloud-lib v0.0.0-20230622122219-feb1c51b50d6/go.mod h1:lT0kVeV0d4HznnITUHbHN6JunpqyL4q9McAYr78YNm4=
+github.com/weka/go-cloud-lib v0.0.0-20230622124207-52128fdad7ff h1:qm9VlAfIFm0eQCI6BdJ2AWcXaHvg/q1g1H7FD7rmb30=
+github.com/weka/go-cloud-lib v0.0.0-20230622124207-52128fdad7ff/go.mod h1:lT0kVeV0d4HznnITUHbHN6JunpqyL4q9McAYr78YNm4=
+github.com/weka/go-cloud-lib v0.0.0-20230622144902-f140365d5662 h1:jhlJM37haptbdxuc8WMpNRhptl132fo27j1d9W+7/+k=
+github.com/weka/go-cloud-lib v0.0.0-20230622144902-f140365d5662/go.mod h1:lT0kVeV0d4HznnITUHbHN6JunpqyL4q9McAYr78YNm4=
+github.com/weka/go-cloud-lib v0.0.0-20230625073119-85d6c80ea543 h1:BwQ4YDjJl/ne+PoGt00wWrGmQZsyacgpzw2Lj/Z4+UQ=
+github.com/weka/go-cloud-lib v0.0.0-20230625073119-85d6c80ea543/go.mod h1:lT0kVeV0d4HznnITUHbHN6JunpqyL4q9McAYr78YNm4=
```
Comment (assafgi) [inline function-app/code/go.sum:103]: you should run `go mod tidy` to clean this

File: variables.tf [hunk 3]
```diff
@@ -172,39 +172,39 @@ variable "container_number_map" {
       frontend = 1
       nvme     = 1
       nics     = 4
-      memory   = "31GB"
+      memory   = ["33GB","31GB"]
```
Comment (assafgi) [inline variables.tf:175]: what is this?

File: functions.tf [hunk 4]
```diff
@@ -144,7 +142,7 @@ resource "azurerm_linux_function_app" "function_app" {
     "INSTALL_URL" =  var.install_weka_url != "" ? var.install_weka_url : "https://$TOKEN@get.weka.io/dist/v1/install/${var.weka_version}/${var.weka_version}"
     "LOG_LEVEL" = var.function_app_log_level
     "SUBNET" = data.azurerm_subnet.subnet.address_prefix
-
+    ADD_FRONTEND_CONTAINERS = var.add_frontend_containers
```
Comment (assafgi) [inline functions.tf:145]: not needed

File: functions.tf [hunk 5]
```diff
@@ -88,15 +88,13 @@ locals {
...
-}
-
-locals {
-  function_code_path     = "${path.module}/function-app/code"
-  function_app_code_hash = md5(join("", [for f in fileset(local.function_code_path, "**") : filemd5("${local.function_code_path}/${f}")]))
+  location                 = data.azurerm_resource_group.rg.location
+  function_app_zip_name    = "${var.function_app_dist}/${var.function_app_version}.zip"
+  weka_sa                  = "${var.function_app_storage_account_prefix}${local.location}"
+  weka_sa_container        = "${var.function_app_storage_account_container_prefix}${local.location}"
+  function_code_path       = "${path.module}/function-app/code"
```
Comment (assafgi) [inline functions.tf:95]: why do you need this?

File: examples/existing_private_network/vars.auto.tfvars (hunk 1 above)
Comment (denise89p) [reply examples/existing_private_network/vars.auto.tfvars:13]: need for user as flag

File: function-app/code/go.sum (hunk 2 above)
Comment (denise89p) [reply function-app/code/go.sum:103]: good to know

File: variables.tf (hunk 3 above)
Comment (denise89p) [reply variables.tf:175]: new memory compute according to num of compute containers

File: functions.tf (hunk 5 above)
Comment (denise89p) [reply functions.tf:95]: to know which compute memory value to take , index 0 mean no frontend,  index 1 mean with frontend

File: variables.tf (hunk 3 above)
Comment (assafgi) [reply variables.tf:175]: so this isn't relevant to frontend PR

File: examples/public_network/vars.auto.tfvars [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
-address_space       = "10.0.0.0/16"
-subnet_prefixes     = "10.0.2.0/24"
-subnet_delegation   = "10.0.1.0/25"
-cluster_name        = "poc"
-instance_type       = "Standard_L8s_v3"
-set_obs_integration = true
-tiering_ssd_percent = 20
-cluster_size        = 6
\ No newline at end of file
+prefix                  = "weka"
```
Comment (assafgi) [inline examples/public_network/vars.auto.tfvars:1]: indentation seems wrong, I assume it is leftover

**Review by assafgi — APPROVED** [review]

**Pull request merged**